rpcs3/Loader/PUP.cpp

```diff
@@ -47,11 +61,18 @@ fs::file pup_object::get_file(u64 entry_id)
 
 bool pup_object::validate_hashes()
 {
+	if (!isValid) return false;
+
 	for (usz i = 0; i < m_file_tbl.size(); i++)
 	{
 		u8 *hash = m_hash_tbl[i].hash;
 		PUPFileEntry file = m_file_tbl[i];
 
+		if (usz size = m_file.size(); size < file.data_offset || size - file.data_offset < file.data_length)
```

Comment:
Wouldn't this be equivalent to
if (m_file.size() < (file.data_offset + file.data_length))

Reply:
Addition can overflow, rememeber that the values are provided by the file itself so it can be iterally any value.

Reply:
Can you add a comment for this?

Reply:
Added.